vars/capsuleSystemTests.groovy

```diff
@@ -0,0 +1,196 @@
+  trysh 'rm ' + testDirectoryPath + '/vegawallet'
+  trysh 'rm ' + testDirectoryPath + '/config_system_tests.hcl'
+}
+
+void call(Map additionalConfig) {
+  def defaultCconfig = [
+    branchDevopsInfra: 'master',
+    branchVegaCapsule: 'main',
+    branchVega: 'develop',
+    branchDataNode: 'develop',
+    branchSystemTests: 'main',
+    branchVegawallet: 'develop',
+    branchProtos: 'develop',
+    branchVegatools: 'develop',
+    
+    systemTestsTestFunction: '',
+    systemTestsTestMark: 'smoke',
+    systemTestsTestDirectory: '',
+    systemTestsDebug: '',
+
+    preapareSteps: {},
+    nodeIdentifier: 'system-tests-capsule',
+    gitCredentialsId: 'vega-ci-bot',
+  ]
+  
+  def config = defaultCconfig + additionalConfig
+
+  print "config: " + config
+
+  node(config.nodeIdentifier) {
+    def testDirectoryPath
+    dir('tests') {
+      testDirectoryPath = sh(returnStdout: true, script: 'pwd').trim()
+    }
+
+    stage('prepare') {
+      cleanup(testDirectoryPath)
+      
+      config.preapareSteps()
+    }
+
+    stage('get source codes') {
+      def repositories = [ 
+        [ name: 'devops-infra', branch: config.branchDevopsInfra ],
+        [ name: 'vegacapsule', branch: config.branchVegaCapsule ],
+        [ name: 'vega', branch: config.branchVega ],
+        [ name: 'data-node', branch: config.branchDataNode ],
+        [ name: 'system-tests', branch: config.branchSystemTests ],
+        [ name: 'vegawallet', branch: config.branchVegawallet ],
+        [ name: 'protos', branch: config.branchProtos ],
+        [ name: 'vegatools', branch: config.branchVegatools ],
+      ]
+
+      parallel repositories.collectEntries{value -> [value.name, { gitClone(value.name, value.branch, value.name, config.gitCredentialsId) }]}
+    }
+
+    stage('build binaries') {
+      def binaries = [
+        [ repository: 'vegacapsule', name: 'vegacapsule', packages: './main.go' ],
+        [ repository: 'vega', name: 'vega', packages: './cmd/vega/' ],
+        [ repository: 'data-node', name: 'data-node', packages: './cmd/data-node/' ],
+        [ repository: 'vegawallet', name: 'vegawallet', packages: './main.go' ],
+      ]
+      
+      parallel binaries.collectEntries{value -> [value.name, { buildGoBinary(value.repository,  testDirectoryPath + '/' + value.name, value.packages) }]}
+    }
+    
+    stage('prepare network') {
+      dir('system-tests') {
+        sh 'cp -r ./vegacapsule/multisig-setup ' + testDirectoryPath
+        sh 'cp ./vegacapsule/capsule_config.hcl ' + testDirectoryPath + '/config_system_tests.hcl'
+      }
+
+      dir ('tests/multisig-setup') {
+        sh 'npm install'
+      }
+      
+      dir ('tests') {
+        withCredentials([
+          usernamePassword(credentialsId: 'github-vega-ci-bot-artifacts', passwordVariable: 'TOKEN', usernameVariable:'USER')
+        ]) {
+          sh 'echo -n "' + TOKEN + '" | docker login https://ghcr.io -u "' + USER + '" --password-stdin'
```

Comment:
Unless we want to stay logged in, a `logout` section is missing.
But then we can solve it differently: add docker config file in Jenkins Agent image.

Reply:
Yess I have added logout step :)